src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v1/oracle/PmQueries.kt

```diff
@@ -152,6 +152,65 @@ val userFiscalCodeCountQuery =
     """
         .trimIndent()
 
+// note: the conversion from Java's OffsetTime to timestamp for the parameters is handled by oracle
+// through jdbc
+val timestampRangePaginatedQuery =
+    """
+        SELECT  
+            pu.FISCAL_CODE, 
+            pu.NOTIFICATION_EMAIL, 
+            pu.SURNAME, 
+            pu.NAME, 
+            pu.USERNAME, 
+            pu.STATUS,
+            pt.CREATION_DATE, 
+            pt.STATUS,
+            pt.ACCOUNTING_STATUS,
+            PP.ORIGIN,
+            pt.AMOUNT, 
+            pt.FEE, 
+            pt.GRAND_TOTAL,
+            pt.rrn,  
+            pt.AUTHORIZATION_CODE,
+            pt.SERVICE_NAME,
+            pp.SUBJECT,
+            ppd.IUV, 
+            ppd.CCP, 
+            ppd.ENTE_BENEFICIARIO, 
+            ppd.ID_DOMINIO,
+            pp2.ID_PSP, 
+            pp2.BUSINESS_NAME, 
+            pp2.ID_CHANNEL
+        FROM AGID_USER.PP_PAYMENT pp
+        LEFT JOIN AGID_USER.PP_TRANSACTION pt 
+            ON pt.FK_PAYMENT = pp.ID 
+        LEFT JOIN AGID_USER.PP_USER pu 
+            ON pu.ID_USER = pt.FK_USER 
+        LEFT JOIN AGID_USER.PP_PAYMENT_DETAIL ppd 
+            ON pp.ID = ppd.PAYMENT_ID 
+        LEFT JOIN AGID_USER.PP_PSP pp2 
+            ON pt.FK_PSP = pp2.ID 
+        WHERE pp.CREATION_DATE BETWEEN ? AND ?
+        OFFSET ? ROWS FETCH NEXT ? ROWS ONLY
+    """
+        .trimIndent()
+
+val timestampRangeCountQuery =
+    """
+        SELECT COUNT(*)
+        FROM AGID_USER.PP_PAYMENT pp
```

Comment:
If code duplication is an issue, some shared query parts can obviously be stored and reused like this (existing working queries will be modified, NRT will be necessary):
```
const val basePmTransactionFromQuery =
    """
    FROM AGID_USER.PP_USER pu 
    LEFT JOIN AGID_USER.PP_TRANSACTION pt 
        ON pu.ID_USER = pt.FK_USER 
    LEFT JOIN AGID_USER.PP_PAYMENT pp 
        ON pt.FK_PAYMENT = pp.ID 
    LEFT JOIN AGID_USER.PP_PAYMENT_DETAIL ppd 
        ON pp.ID = ppd.PAYMENT_ID 
    LEFT JOIN AGID_USER.PP_PSP pp2 
        ON pt.FK_PSP = pp2.ID
"""

val timestampRangeCountQuery =
    """
    SELECT COUNT(*)
    $basePmTransactionFromQuery
    WHERE pp.CREATION_DATE BETWEEN ? AND ?
"""
        .trimIndent()
```

Reply:
For now it's fine, we'll do a refactoring when we have the final version